src/algebra/module/basic.lean

```diff
@@ -478,6 +478,24 @@ class no_zero_smul_divisors (R M : Type*) [has_zero R] [has_zero M] [has_scalar
 
 export no_zero_smul_divisors (eq_zero_or_eq_zero_of_smul_eq_zero)
 
+/-- A `no_zero_smul_divisors` instance on the codomain of a type coercing to functions induces
+one on that type, given appropriate properties of the types and scalar actions. -/
+lemma function.injective.no_zero_smul_divisors {R F M N : Type*} [has_zero R] [has_zero F]
+  [has_zero N] [has_scalar R F] [has_scalar R N] [no_zero_smul_divisors R N]
+  [has_coe_to_fun F (λ _, M → N)] (hi : function.injective (coe_fn : F → M → N))
```

Comment:
Are you sure you can't just make this about
```suggestion
  (f : F → N) (hi : function.injective f)
```
I'll have a go in gitpod later

Reply:
[See below](https://github.com/leanprover-community/mathlib/pull/10234/files#r745470277)